docs/one-click-integrations/bitbucket-integration.md

```diff
@@ -0,0 +1,170 @@
+# Bitbucket integration
+
+!!! note
+    (CUSTOMER FACING MESSAGE NEEDED) We have launched Bitbucket integration to receive your feedback. Some metrics are not available yet.
+
+Pulse integrates with Bitbucket Cloud to receive data about changes and deployments, necessary to calculate the metrics:
+
+-   [Lead time for changes](../metrics/accelerate.md#lead-time-for-changes)
+
+-   [Deployment frequency](../metrics/accelerate.md#deployment-frequency)
+
+## Setting up the Bitbucket integration
+
+!!! note
+    -   The Bitbucket integration connects to a specific [workspace](https://support.atlassian.com/bitbucket-cloud/docs/what-is-a-workspace/). You must be the **owner** of that Bitbucket workspace to proceed with the integration setup.
+    -   [See below](#bb-permissions) the permissions that Pulse requires from your Bitbucket account.  
+
+To set up the Bitbucket integration:
+
+1.  On Pulse, [expand **Integrations** and select **Bitbucket**](https://app.pulse.codacy.com/integrations/bitbucket){: target="_blank"}.
+
+1.  Click **Install Bitbucket App** and follow the instructions on the Bitbucket UI to install the app on your account.
+
+    (IMAGE)
+
+1.  Wait until you get a confirmation that Pulse successfully created the integration and the webhook on Bitbucket.
+            
+    (IMAGE)
+
+    If there was an error please [contact support](mailto:pulsesupport@codacy.com).
+
+1.  Choose the workspace that you want to connect the integration. You will only be able to select a workspace that you are the owner.
+
+    (IMAGE)
+
+    If there was an error please [contact support](mailto:pulsesupport@codacy.com).
+
+1.  Click **Complete setup**.
+
+## Automatic deployment detection using merged pull requests {: id="deployment-detection-merged-pr"}
+
+!!! note
+    Pulse will use your merged pull requests to automatically detect your deployments. Other strategies to detect deployments will be available with a future version of Pulse.
+
+-   Pulse considers a deployment every merged pull request that **targets the default branch** of the repository.
+-   The deployment date is the timestamp when the corresponding pull request is merged.
+-   The set of changes in a deployment is the list of commits in the corresponding pull request. Pulse correctly tracks your changes even if you [squash or fast-forward the commits when merging the pull request](https://support.atlassian.com/bitbucket-cloud/docs/merge-a-pull-request/#Merge-strategies), since Pulse processes all the original commits before any changes to the Git history.
+-   Pulse associates all Bitbucket teams of the author of a merged pull request with the corresponding deployment, excluding teams with less than two members. Pulse only takes changes to Bitbucket teams into account on pull requests merged after those changes.
+
+## Collected data
```

Comment:
@afsmeira Can you please validate the collected data for Bitbucket integration? I based it on the GitHub content, but maybe the mapping to the Pulse data model needs some tweaks.

Reply:
<img width="655" alt="Screenshot 2022-12-14 at 11 59 04" src="https://user-images.githubusercontent.com/6381457/207589842-c51c80d2-e530-4b89-bcd9-c5268ea829a9.png">

☝️ These are the permissions we request.

- **Email** - to get the email of the user doing the integration.
- **Account** (requires Email) - to list workspaces, user groups and check workspace membership.
- **Workspace membership** - I actually don't think this is required. I will investigate further.
- **Repositories** - get repository default branch, list commits.
- **Pull request** (requires repository) - list pull requests and their activity.
- **Webhooks** - to created and delete workspace webhooks.

Regarding webhooks, we subscribe to:
- Pull request events
    - Created
    - Updated
    - Fulfilled
    - Rejected
    - Approved
    - Unapproved
    - Changes request created
    - Changes request removed
- Repository events
    - Pushed reference (we're only interested tags)
    - Transfered (repository transfered to the selected workspace, so that we can start historical data import)
    - Forked (repository created in the selected workspace as a fork of another repository, so that we can start historical data import)
    
The permissions and events will be requested and subscribed to since V1 but we might not use them all in V1.

Reply:
Updated permissions table.